### PR TITLE
Notebooks updated to pytorch==1.0.0

### PR DESCRIPTION
**Pull request by avinashsai**
I updated few notebooks to pytorch 1.0 with necessary additions and  added watermark at the end of the notebooks.

### CODE REVIEW

**Comment by rasbt** [comment]
Thanks! Now on a second glance, the fix with the NLL loss seems redundant to me. I think CrossEntropyLoss automatically uses log_softmax if you supply it with the logits directly, correct?

**Comment by avinashsai** [comment]
Yes log_softmax+NLL loss == CrossEntropyLoss

**Comment by avinashsai** [comment]
Generally we don't apply any activation function to output layer if we have less no of classes. If there are more classes we normalize outputs with softmax activation. As I said softmax involves heavier computation if there are larger values. log_softmax is very efficiently implemented in pytorch. Hence I always prefer using log_softmax+NLL loss. Comparing log values is easier. 
It is just a matter of choice. If you want me to revert those I will keep CrossEntropy as loss function.

**Comment by rasbt** [comment]
I agree with you. I usually use CrossEntropyLoss though directly with logits, which is analogous.

I actually noticed that by accident a year back when I was new to PyTorch, 

```python
def mod_cross_entropy(logits, targets, num_classes, device, reduction='elementwise_mean'):
    #smax = F.softmax(logits, dim=1)
    
    weights = ...
    expanded_targets = (torch.arange(num_classes).to(device) == targets[:, None]).float()
    #val = - torch.sum(torch.log(smax) * expanded_targets * weights[:, np.newaxis], dim=1)
    val = - torch.sum(F.log_softmax(logits, dim=1) * expanded_targets * weights[:, None], dim=1)
    if reduction == 'elementwise_mean':
        return torch.mean(val)
    elif reduction == 'elementwise_sum':
        return torch.sum(val)
    elif reduction == 'none':
        return val
    else:
        raise NotImplemented
 ```

and when I used the code commented out instead (i.e., log(softmax) instead of log_softmax) I got first the same results but after ~100 epochs, I got NaNs presumably to numerical instability ... which is how i discovered that there was a log_softmax.

Sorry about that, but I think the previous version is maybe more generally useful as the model returns both the logits and the probabilities (which are useful in different applications contexts to get the confidence along with the label, i.e., in inference mode after training -- ok one could also use exp[softmax])

However, with that change to returning log-softmax and then returning logits would maybe be a bit redundant. So, maybe it's better to keep it as is. Sorry for the trouble, that's an interesting discussion though.

**Comment by avinashsai** [comment]
Okay then I will revert those changes.
